java/core/src/java/org/apache/orc/impl/ColumnStatisticsImpl.java

```diff
@@ -517,10 +519,14 @@ public int hashCode() {
 
   protected static final class StringStatisticsImpl extends ColumnStatisticsImpl
       implements StringColumnStatistics {
+    public static final int MAX_STRING_LENGTH_RECORDED = 1024;
```

Comment:
We should probably use the number of bytes instead of the number of characters. I'd propose that we always truncate a whole number of characters, with a maximum of 1024 bytes.

For ASCII strings, it will have the same result, but larger characters may end up at 1021 to 1024 bytes long.